pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -336,6 +336,9 @@ func (r *ReconcileJaeger) apply(ctx context.Context, jaeger v1.Jaeger, str strat
 		if err := r.applyRoutes(ctx, jaeger, str.Routes()); err != nil {
 			return jaeger, tracing.HandleError(err, span)
 		}
+		if err := r.applyConsoleLinks(ctx, jaeger, r.updateHref(ctx, jaeger, str.ConsoleLinks())); err != nil {
```

Comment:
The controller shouldn't know how to transform the existing console link into desired. In other words, the `updateHref` logic shouldn't happen here.

The way our operator works, this code here should only call `applyConsoleLinks` passing `str.ConsoleLinks()` as parameter, which builds the list of desired console links. The `applyConsoleLinks` will then get this list over to the inventory, which will compare the existing with the desired and build a map of operations: what should be created, what should be updated, and what should be deleted. The inventory should take care of carrying over the value from the desired to the existing, in case of an update (which is what your `updateHref` is doing). Take a look at what we are doing with the secret or deployment. The batch job isn't a good example.